newrelic/api/opentelemetry.py

```diff
@@ -216,10 +217,17 @@ def set_attributes(self, attributes):
             self.set_attribute(key, value)
 
     def _set_attributes_in_nr(self, otel_attributes=None):
-        if not (otel_attributes and hasattr(self, "nr_trace") and self.nr_trace):
+        if not otel_attributes or not getattr(self, "nr_trace"):
             return
+        
+        # If these attributes already exist in NR's intrinsic or agent
+        # attributes, keep the attributes in the OTel span, but do not
+        # add them to NR's user attributes to avoid sending the same
+        # data multiple times.
+        combined_attrs = {**self.nr_trace.agent_attributes, **self.nr_trace.user_attributes}
```

Comment:
You might be able to do this logic more simply with a set:
```suggestion
        otel_user_attrs = (set(self.nr_trace.agent_attributes) | set(self.nr_trace.user_attributes)) & set(otel_attributes)
        self.nr_trace.add_custom_attributes({key: otel_attributes[key] for key in otel_user_attrs}) 
```